main.go

```diff
@@ -311,3 +321,23 @@ func run(ctrlConfig spirev1alpha1.ControllerManagerConfig, options ctrl.Options)
 
 	return nil
 }
+
+func autoDetectClusterDomain() string {
+	cname, err := net.LookupCNAME(k8sDefaultService)
+	if err != nil {
+		setupLog.Error(err, "unable to autodetect cluster domain")
+		return ""
+	}
+
+	re := regexp.MustCompile(regexp.QuoteMeta(k8sDefaultService) + `\.(.*)\.`)
+	matches := re.FindStringSubmatch(cname)
+
+	var clusterDomain string
+	if len(matches) == 2 {
```

Comment:
This len check is a little confusing since it will either match, or it won't. So you will either have matches == nil (didn't match) or matches[0] = whole match with matches[1] = first suggroup match. This also means that the matches logged in the else case will always be empty.

Reply:
N/A anymore since regex removed.